core/chains/evm/logpoller/log_poller.go

```diff
@@ -813,6 +816,22 @@ func (lp *logPoller) PollAndSaveLogs(ctx context.Context, currentBlockNumber int
 	}
 }
 
+func (lp *logPoller) saveLogsInTx(ctx context.Context, fn func(tx pg.Queryer) error) error {
+	err := lp.orm.q.WithOpts(pg.WithParentCtx(ctx)).Transaction(func(tx pg.Queryer) error {
+		return fn(tx)
+	})
+	if err != nil {
+		return err
+	}
+
+	select {
+	case lp.notify <- struct{}{}:
```

Comment:
Doesn't this notify on every single log? That's going to be very noisy and affect listeners that aren't related. Presumably we would want to notify on a particular topic?

Reply:
Good point. I changed the signature of LogPoller.Notify to only notify on a specified topic index + value pair.